core/offchain.go

```diff
@@ -198,8 +203,14 @@ func (bc *BlockChain) CommitOffChainData(
 			); err != nil {
 				return NonStatTy, err
 			}
-			if err := bc.DeletePendingSlashingCandidates(); err != nil {
-				return NonStatTy, err
+			records := slash.Records{}
+			if s := header.Slashes(); len(s) > 0 {
+				if err := rlp.DecodeBytes(s, &records); err != nil {
```

Comment:
should be lenient on here too. If the slash can't be deleted, they should be deleted at next proposal anyway as they will be invalid/duplicate.

Reply:
Hmm, so remove error return on both decode & DeleteFromPending?

Reply:
yup.